subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -242,7 +242,7 @@ static void handle_wifi_disconnect_result(struct net_mgmt_event_callback *cb)
 	const struct shell *sh = context.sh;
 
 	if (context.disconnecting) {
-		if (status->status) {
+		if (status->status != WIFI_STATUS_CONN_LAST_STATUS) {
```

Comment:
```suggestion
		if (status->status != WIFI_REASON_DISCONN_UNSPECIFIED) {
```

Even this isn't quite right, we need to have a separate (status, reason), and if status is failure then reason comes in to play.